src/main/flight/pid.c

```diff
@@ -740,6 +741,48 @@ float pidGetAirmodeThrottleOffset(void)
 }
 #endif
 
+// when new Rx data is received, update the maxDeflectionAbs value, so we don't do this every PID loop
+static float maxDeflectionAbs = 0.0f;
+float sendMaxDeflectionAbs(float maxRcDeflectionAbs)
+{
+    maxDeflectionAbs = fmaxf(maxRcDeflectionAbs, mixerGetRcThrottle());
+    return maxDeflectionAbs;
+}
+
+static void disarmOnImpact(const int axis)
+{
+    // if both sticks are inside 20% of the stick threshold, enable auto-disarm on impact
+    // will not disarm on gentle landings
+    // value should be highe enough to avoid unwanted disarms in the air on throttle chops
+    // note that unlike GPS code, this just compares one axis at a time, to reduce CPU load
+    // for the Z axis, 1G gets added while 'flat', but that won't matter much given we are looking around 8G mostly
+    float accMagnitude = fabsf(acc.accADC[axis]) * acc.dev.acc_1G_rec;
```

Comment:
the acceleration's magnitude is the square root of the squared sums of all axes.
```suggestion
    float accMagnitude = sq(acc.accADC[X]) + sq(acc.accADC[Y]) + sq(acc.accADC[Z]);
    accMagnitude = sqrtf(accMagnitude);
    accMagnitude *= acc.dev.acc_1G_rec;
```

Reply:
True, but this code is called three times per axis - 9 times per pid loop.
And I think that if we look at each axis individually, the worst case is returning 0.7 when the vector is 1.0.  That's not a big deal I think.
Otherwise I should take the disarm code out of the per-axis loop.

Reply:
Thanks.  Solved by doing the calculation once per PID loop.